Read named arguments from a file, SCM or HTTP(S) endpoint

For appliances it is more common to use someone else's senza template and end up providing your own parameters.

Sometimes it could make more sense to just feed the values for those parameters directly from a file or directly from some Git or HTTP endpoint.

For ex., to use the stups-cassandra appliance, directly from https://github.com/zalando/stups-cassandra, one could write a personal stups-cassandra-_myteam_.yaml with parameter values specific to _myteam_:

    EtcdDomain: etcd.myteam.example.com
    ScalyrKey: myteam-specific-scalyr-api-key
    ApplicationId: myteam-stups-cassandra-according-to-yourturn

Such parameter files could be used with a special notation like:

    senza create definition.yaml version @stups-cassandra-myteam.yaml

or

    senza create --params stups-cassandra-myteam.yaml definition.yaml version

Along with https://github.com/zalando-stups/senza/issues/99 this could even become something like:

    senza create \
        --params https://raw.githubusercontent.com/myself/project/master/stups-cassandra-myteam.yaml  \
        https://raw.githubusercontent.com/zalando/stups-cassandra/master/stups-cassandra.yaml \
        cluster1

We're now doing something very similar to this using the strategy i described in #155 

That approach still makes the templates long and ugly, not to mention that it allowed me to find the 1024 chars limits for a stack's description:
```
botocore.exceptions.ClientError: An error occurred (ValidationError) when calling the CreateStack operation: Template format error: 'Description' length is greater than 1024.
Makefile:23: recipe for target 'deploy' failed
```

--- a/senza/cli.py
+++ b/senza/cli.py
@@ -403,7 +403,15 @@ def read_parameter_file(parameter_file):
         raise click.UsageError('Can\'t read parameter file "{}"'.format(parameter_file))
 
     try:
-        cfg = yaml.safe_load(ymlfile)
+        url = (parameter_file if '://' in parameter_file
+               else 'file://{}'.format(quote(os.path.abspath(parameter_file))))
+
+        response = urlopen(url)
+    except URLError:
+        raise click.UsageError('"{}" not found'.format(parameter_file))
+
+    try:
+        cfg = yaml.safe_load(response.read())
         for key, val in cfg.items():
             paras.append("{}={}".format(key, val))
     except yaml.YAMLError as e:

--- a/senza/cli.py
+++ b/senza/cli.py
@@ -397,10 +397,6 @@ def parse_args(input, region, version, parameter, account_info):
 
 def read_parameter_file(parameter_file):
     paras = []
-    try:
-        ymlfile = open(parameter_file, 'r')
-    except OSError:
-        raise click.UsageError('Can\'t read parameter file "{}"'.format(parameter_file))
 
     try:
         url = (parameter_file if '://' in parameter_file
@@ -408,7 +404,7 @@ def read_parameter_file(parameter_file):
 
         response = urlopen(url)
     except URLError:
-        raise click.UsageError('"{}" not found'.format(parameter_file))
+        raise click.UsageError('Can\'t read parameter file "{}"'.format(parameter_file))
 
     try:
         cfg = yaml.safe_load(response.read())

